### PR TITLE
Login-And-Registration-Flow

### PR DESCRIPTION
**Pull request by IvayloSlavchev**

### CODE REVIEW

**Review by petya0111 — APPROVED** [review]
Looking good! ✓

File: client/src/components/Header/Header.js [hunk 1]
```diff
@@ -94,12 +102,17 @@ const Header = () => {
...
-
+              {
+                username == 'Sign Up' ?  <Link to='/sign' style={textDecorationObject}>
+                <Button className='header-buttons'><AccountCircleRoundedIcon color='disabled' fontSize='large' />
+                  <Typography className='pages'>{username}</Typography>
+                </Button>
+              </Link>  :  <Link to='/user' style={textDecorationObject}>
+                <Button className='header-buttons'><AccountCircleRoundedIcon color='disabled' fontSize='large' />
+                  <Typography className='pages'>{username}</Typography>
+                </Button>
```
Comment (Arabadzhiew) [inline client/src/components/Header/Header.js:120]: This ternary will be easier to read if the two JSX values are declared as constants and accesed trough them in the expression.

File: client/src/components/SignUp/Inputs.js [hunk 2]
```diff
@@ -86,7 +91,10 @@ export default function SignUp() {
                     <Grid item xs={12}>
                         <Typography className='tearms-and-conditions'>By signing in you accept Exordium Terms and Conditions <Link to="#" sx={{ cursor: 'pointer' }}>Learn more</Link></Typography>
                     </Grid>
-                    <Button type='submit' data-testid='submit-button' className='create-account-button' sx={{ textTransform: 'none' }}>Create account</Button>
+                    {
+                        !isRegistrated ? <Button type='submit' data-testid='submit-button' className='create-account-button' sx={{ textTransform: 'none' }}>Create account</Button>
+                        : <Navigate to='/login' type='submit'><Button type='submit' data-testid='submit-button' className='create-account-button' sx={{ textTransform: 'none' }}>Create account</Button></Navigate>
+                    }
```
Comment (Arabadzhiew) [inline client/src/components/SignUp/Inputs.js:97]: Same as above. Export the JSX in constants and access it trough them.

**Review by JordanRad — APPROVED** [review]

File: client/src/components/Header/Header.js (hunk 1 above)
Comment (IvayloSlavchev) [reply client/src/components/Header/Header.js:120]: Fixed

File: client/src/components/SignUp/Inputs.js (hunk 2 above)
Comment (IvayloSlavchev) [reply client/src/components/SignUp/Inputs.js:97]: Fixed

**Review by fbozhkov — CHANGES_REQUESTED** [review]
Please change the message "Password should be at least 8 characters long" to show all the requirements for the password ( at least one capital letter, and a number). Right now it isn't very clear what the password should include and leads to confusion if the user gives a pasword like '12345678' for example.

File: client/src/components/SignIn/SignIn.js [hunk 3]
```diff
@@ -107,8 +113,10 @@ export default function SignIn() {
                                     onChange={(event) => { setUserPassword(event.target.value); }}
                                 /> : null
                             }
-                            <Box className='button-class'>
-                                <Button className='log-in-account-button' type='submit'>Continue</Button>
+                            <Box className='button-class'> 
+                                {
+                                    !isLogged ? notAcceptedLoginRequest : loginButton
```
Comment (dplamenov) [inline client/src/components/SignIn/SignIn.js]: it is better to remove `!`

isLogged ? loginButton : notAcceptedLoginRequest

Comment (IvayloSlavchev) [reply client/src/components/SignIn/SignIn.js]: Done

**Comment by IvayloSlavchev** [comment]
> Please change the message "Password should be at least 8 characters long" to show all the requirements for the password ( at least one capital letter, and a number). Right now it isn't very clear what the password should include and leads to confusion if the user gives a pasword like '12345678' for example.

Done

**Review by fbozhkov — APPROVED** [review]

**Pull request merged**